pennylane/ftqc/graph_state_preparation.py

```diff
@@ -0,0 +1,166 @@
+        QubitGraph<square, (1, 0)>: ──Y─╰X────│──╭●─┤  Probs
+        QubitGraph<square, (1, 1)>: ──Y───────╰X─╰X─┤  Probs
+    """
+
+    def __init__(
+        self,
+        graph: Union[nx.Graph, QubitGraph],
+        qubit_ops: Operation = qml.H,
+        entanglement_ops: Operation = qml.CZ,
+        wires: Wires = None,
+    ):
+        self.hyperparameters["graph"] = graph
+        self.hyperparameters["qubit_ops"] = qubit_ops
+        self.hyperparameters["entanglement_ops"] = entanglement_ops
+
+        if isinstance(graph, QubitGraph):
+            if wires is not None and set(wires) != set(graph.graph):
+                raise ValueError("Please ensure wires objects match labels in QubitGraph")
+            super().__init__(wires=wires if wires is not None else set(graph.graph))
+        else:
+            if wires is None:
+                raise ValueError("Please ensure wires is specified.")
+            if wires is not None and set(wires) != set(graph):
+                raise ValueError("Please ensure wires objects match labels in graph")
+            super().__init__(wires=wires)
+
+    def label(
+        self, decimals: int = None, base_label: str = None, cache: dict = None
+    ):  # pylint: disable=unused-argument
+        r"""How the graph state preparation is represented in diagrams and drawings.
+
+        Args:
+            decimals: If ``None``, no parameters are included. Else, how to round
+                the parameters. Required to match general call signature. Not used.
+            base_label: overwrite the non-parameter component of the label.
+                Required to match general call signature. Not used.
+            cache: dictionary that carries information between label calls in the
+                same drawing. Required to match general call signature. Not used.
+
+        Returns:
+            str: label to use in drawings
+        """
+        return f"GraphStatePrep({self.hyperparameters['qubit_ops'](wires=0).name}, {self.hyperparameters['entanglement_ops'].name})"
+
+    def __repr__(self):
+        """Method defining the string representation of this class."""
+        return f"GraphStatePrep({self.hyperparameters['qubit_ops'](wires=0).name}, {self.hyperparameters['entanglement_ops'].name})"
+
+    def decomposition(self) -> list["Operator"]:
+        r"""Representation of the operator as a product of other operators.
+
+        Returns:
+            list[Operator]: decomposition of the operator
+        """
+        return self.compute_decomposition(wires=self.wires, **self.hyperparameters)
+
+    @staticmethod
+    def compute_decomposition(
+        wires: Wires,
+        graph: Union[nx.Graph, QubitGraph],
+        qubit_ops: Operation = qml.H,
+        entanglement_ops: Operation = qml.CZ,
+    ):  # pylint: disable=arguments-differ, unused-argument
+        r"""Representation of the operator as a product of other operators (static method).
+        .. note::
```

Comment:
Need a new line before `.. note::` for Sphinx to render it correctly:

```suggestion

        .. note::
```